namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -225,4 +257,101 @@ class HttpControlService(storage: DtabStore) extends Service[Request, Response]
       case Throw(_: DtabNamespaceDoesNotExistException) => Future.value(Response(Status.NotFound))
       case Throw(_) => Future.value(Response(Status.InternalServerError))
     }
+
+  private[this] def streamingResp[T](values: Event[T], render: (T, Closable) => Buf): Future[Response] = {
+    val resp = Response()
+    resp.setChunked(true)
+    val writer = resp.writer
+    var closable: Closable = null
+    var writeFuture: Future[Unit] = Future.Unit
+    closable = values.respond { t =>
+      writeFuture = writeFuture.before {
+        writer.write(render(t, closable)).onFailure { _ =>
+          closable.close()
+        }
+      }
+    }
+    Future.value(resp)
+  }
+
+  private[this] val bindingCacheMu = new {}
+  private[this] var bindingCache: Map[(String, Path), Activity[NameTree[Name.Bound]]] = Map.empty
+  private[this] def getBind(ns: String, path: Path): Activity[NameTree[Name.Bound]] =
+    bindingCacheMu.synchronized {
+      val key = (ns, path)
+      bindingCache.get(key) match {
+        case Some(act) => act
+        case None =>
+          val act = namers(ns).bind(Dtab.empty, path)
+          bindingCache += (key -> act)
+          act
+      }
+    }
```

Comment:
To wit: I think you could do something like (assuming we can take on a dependency on `scala-java8-compat`; otherwise it would be slightly more verbose)

```scala
private[this] val bindingCache =
  new ConcurrentHashMap[(String, Path), Activity[NameTree[Name.Bound]]]
  
private[this] def getBind(ns: String, path: Path): Activity[NameTree[Name.Bound]] = {
  val fn: ((String, Path)) => Activity[NameTree[Name.Bound]] =
      (ns: String, path: Path) => namers(ns).bind(Dtab.empty, path)
  bindingCache.computeIfAbsent((ns, path), fn.asJava)
}
```